### PR TITLE
DOCSP-35982: Count usage example

### PR DESCRIPTION
**Pull request by norareidy**
JIRA - https://jira.mongodb.org/browse/DOCSP-35982
Staging - https://preview-mongodbnorareidy.gatsbyjs.io/laravel/DOCSP-35982/usage-examples/count/

### Checklist

- [ ] Add tests and ensure they pass
- [ ] Add an entry to the CHANGELOG.md file
- [ ] Update documentation for new features

### CODE REVIEW

File: docs/usage-examples/count.txt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+   :keywords: total, code example
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+You can count the number of documents returned by a query by calling the ``where()`` and
+``count()`` methods on an object collection or a query builder.
```
Comment (GromNaN) [inline docs/usage-examples/count.txt]: Think Laravel name the document instance "model".

```suggestion
``count()`` methods on a collection of models, or a query builder.
```

File: docs/includes/usage-examples/CountTest.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            ],
+            [
+                'title' => 'Million Dollar Mermaid',
+                'genres' => ['Biography', 'Drama', 'Musical'],
+            ],
+        ]);
+
+        // begin-count
+        $biographies = Movie::where('genres', 'Biography')
+            ->count();
```
Comment (GromNaN) [inline docs/includes/usage-examples/CountTest.php]: The 2nd example for "a collection of object" would be:

```php
$biographies = Movie::where('genres', 'Biography')->get();
$count = $biographies->count();
```

File: docs/includes/usage-examples/CountTest.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+        echo 'Number of documents: ' . $biographies;
+        // end-count
+
+        $this->assertEquals(2, $biographies);
+        $this->expectOutputString('Number of documents: 2');
```
Comment (GromNaN) [inline docs/includes/usage-examples/CountTest.php]: The name `$count` is more for the variable that contains a number of documents.

```suggestion
        $count = Movie::where('genres', 'Biography')
            ->count();

        echo 'Number of documents: ' . $count;
        // end-count

        $this->assertEquals(2, $count);
        $this->expectOutputString('Number of documents: 2');
```

File: docs/usage-examples/count.txt [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+- Uses the ``Movie`` Eloquent model to represent the ``movies`` collection in the
+  ``sample_mflix`` database
+- Counts the documents from the ``movies`` collection that match a query filter
+- Prints the matching document count
+
+The example calls the following methods on the ``Movie`` model:
+
+- ``where()``: matches documents in which the value of the ``genres`` field includes ``"Biography"``.
+- ``count()``: counts the number of matching documents. This method returns an integer value.
```
Comment (mongoKart) [inline docs/usage-examples/count.txt]: [Style Guide](https://www.mongodb.com/docs/meta/style-guide/style/punctuation/#colons:~:text=In%20a%20list%20item%2C%20if%20you%20need%20to%20separate%20an%20initial%20term%20or%20phrase%20from%20the%20information%20that%20follows%20it%2C%20use%20a%20colon.%20For%20example%3A)

```suggestion
- ``where()``: Matches documents in which the value of the ``genres`` field includes ``"Biography"``.
- ``count()``: Counts the number of matching documents. This method returns an integer value.
```

**Review by mongoKart — APPROVED** [review]
LGTM + 1 fix

File: docs/includes/usage-examples/CountTest.php (hunk 2 above)
Comment (norareidy) [reply docs/includes/usage-examples/CountTest.php]: Does the `where()` method return a collection of objects? If so, it seems (from my understanding) accurate to say that I'm calling `count()` on a "collection of objects" if I chain `count()` right after `where()`. If not, I'll change the other code examples that mention a "collection of objects" as well to use `get()`.

Comment (GromNaN) [reply docs/includes/usage-examples/CountTest.php]: `where()` returns a query builder. Calling `->get()` runs the find query and returns the collection of model instances.

**Review by GromNaN — APPROVED** [review]

**Pull request merged**